aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -388,7 +388,7 @@ static inline Tensor to_impl(
     c10::optional<c10::MemoryFormat> optional_memory_format) {
 
   // fast path
-  if (to_will_alias(self, dtype, layout, device, copy, optional_memory_format)) {
+  if (to_will_alias(self, dtype, layout, device, copy, optional_memory_format) && self.const_data_ptr()) {
```

Comment:
Add a comment for const_data_ptr(), it's not trivial this is for fakeTensor/functionalTensor.

Reply:
Also this is a proxy check for under FakeMode? 
maybe we can turn this into an explicit check for under FakeMode?